// growtopia.js/dist/util/Constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TileExtraTypes = exports.TileActionTypes = exports.TileCategoryFlags = exports.TilePropertiesFlags = exports.TileOptionsFlags = exports.TileFlags = exports.TankTypes = exports.PacketTypes = exports.VariantTypes = void 0;
/**
 * Types for each Argument.
 */
var VariantTypes;
(function (VariantTypes) {
    VariantTypes[VariantTypes["NONE"] = 0] = "NONE";
    VariantTypes[VariantTypes["FLOAT_1"] = 1] = "FLOAT_1";
    VariantTypes[VariantTypes["STRING"] = 2] = "STRING";
    VariantTypes[VariantTypes["FLOAT_2"] = 3] = "FLOAT_2";
    VariantTypes[VariantTypes["FLOAT_3"] = 4] = "FLOAT_3";
    VariantTypes[VariantTypes["UNSIGNED_INT"] = 5] = "UNSIGNED_INT";
    VariantTypes[VariantTypes["SIGNED_INT"] = 9] = "SIGNED_INT";
})(VariantTypes || (exports.VariantTypes = VariantTypes = {}));
/**
 * Growtopia Packet Types
 */
var PacketTypes;
(function (PacketTypes) {
    PacketTypes[PacketTypes["UNK"] = 0] = "UNK";
    PacketTypes[PacketTypes["HELLO"] = 1] = "HELLO";
    PacketTypes[PacketTypes["STR"] = 2] = "STR";
    PacketTypes[PacketTypes["ACTION"] = 3] = "ACTION";
    PacketTypes[PacketTypes["TANK"] = 4] = "TANK";
    PacketTypes[PacketTypes["ERROR"] = 5] = "ERROR";
    PacketTypes[PacketTypes["TRACK"] = 6] = "TRACK";
    PacketTypes[PacketTypes["CLIENT_LOG_REQ"] = 7] = "CLIENT_LOG_REQ";
    PacketTypes[PacketTypes["CLIENT_LOG_RES"] = 8] = "CLIENT_LOG_RES";
})(PacketTypes || (exports.PacketTypes = PacketTypes = {}));
/**
 * Growtopia Tank Packet Types
 */
var TankTypes;
(function (TankTypes) {
    TankTypes[TankTypes["STATE"] = 0] = "STATE";
    TankTypes[TankTypes["CALL_FUNCTION"] = 1] = "CALL_FUNCTION";
    TankTypes[TankTypes["UPDATE_STATUS"] = 2] = "UPDATE_STATUS";
    TankTypes[TankTypes["TILE_CHANGE_REQUEST"] = 3] = "TILE_CHANGE_REQUEST";
    TankTypes[TankTypes["SEND_MAP_DATA"] = 4] = "SEND_MAP_DATA";
    TankTypes[TankTypes["SEND_TILE_UPDATE_DATA"] = 5] = "SEND_TILE_UPDATE_DATA";
    TankTypes[TankTypes["SEND_TILE_UPDATE_DATA_MULTIPLE"] = 6] = "SEND_TILE_UPDATE_DATA_MULTIPLE";
    TankTypes[TankTypes["TILE_ACTIVATE_REQUEST"] = 7] = "TILE_ACTIVATE_REQUEST";
    TankTypes[TankTypes["TILE_APPLY_DAMAGE"] = 8] = "TILE_APPLY_DAMAGE";
    TankTypes[TankTypes["SEND_INVENTORY_STATE"] = 9] = "SEND_INVENTORY_STATE";
    TankTypes[TankTypes["ITEM_ACTIVATE_REQUEST"] = 10] = "ITEM_ACTIVATE_REQUEST";
    TankTypes[TankTypes["ITEM_ACTIVATE_OBJECT_REQUEST"] = 11] = "ITEM_ACTIVATE_OBJECT_REQUEST";
    TankTypes[TankTypes["SEND_TILE_TREE_STATE"] = 12] = "SEND_TILE_TREE_STATE";
    TankTypes[TankTypes["MODIFY_ITEM_INVENTORY"] = 13] = "MODIFY_ITEM_INVENTORY";
    TankTypes[TankTypes["ITEM_CHANGE_OBJECT"] = 14] = "ITEM_CHANGE_OBJECT";
    TankTypes[TankTypes["SEND_LOCK"] = 15] = "SEND_LOCK";
    TankTypes[TankTypes["SEND_ITEM_DATABASE_DATA"] = 16] = "SEND_ITEM_DATABASE_DATA";
    TankTypes[TankTypes["SEND_PARTICLE_EFFECT"] = 17] = "SEND_PARTICLE_EFFECT";
    TankTypes[TankTypes["SET_ICON_STATE"] = 18] = "SET_ICON_STATE";
    TankTypes[TankTypes["ITEM_EFFECT"] = 19] = "ITEM_EFFECT";
    TankTypes[TankTypes["SET_CHARACTER_STATE"] = 20] = "SET_CHARACTER_STATE";
    TankTypes[TankTypes["PING_REPLY"] = 21] = "PING_REPLY";
    TankTypes[TankTypes["PING_REQUEST"] = 22] = "PING_REQUEST";
    TankTypes[TankTypes["GOT_PUNCHED"] = 23] = "GOT_PUNCHED";
    TankTypes[TankTypes["APP_CHECK_RESPONSE"] = 24] = "APP_CHECK_RESPONSE";
    TankTypes[TankTypes["APP_INTEGRITY_FAIL"] = 25] = "APP_INTEGRITY_FAIL";
    TankTypes[TankTypes["DISCONNECT"] = 26] = "DISCONNECT";
    TankTypes[TankTypes["BATTLE_JOIN"] = 27] = "BATTLE_JOIN";
    TankTypes[TankTypes["BATTLE_EVEN"] = 28] = "BATTLE_EVEN";
    TankTypes[TankTypes["USE_DOOR"] = 29] = "USE_DOOR";
    TankTypes[TankTypes["SEND_PARENTAL"] = 30] = "SEND_PARENTAL";
    TankTypes[TankTypes["GONE_FISHIN"] = 31] = "GONE_FISHIN";
    TankTypes[TankTypes["STEAM"] = 32] = "STEAM";
    TankTypes[TankTypes["PET_BATTLE"] = 33] = "PET_BATTLE";
    TankTypes[TankTypes["NPC"] = 34] = "NPC";
    TankTypes[TankTypes["SPECIAL"] = 35] = "SPECIAL";
    TankTypes[TankTypes["SEND_PARTICLE_EFFECT_V2"] = 36] = "SEND_PARTICLE_EFFECT_V2";
    TankTypes[TankTypes["ACTIVE_ARROW_TO_ITEM"] = 37] = "ACTIVE_ARROW_TO_ITEM";
    TankTypes[TankTypes["SELECT_TILE_INDEX"] = 38] = "SELECT_TILE_INDEX";
    TankTypes[TankTypes["SEND_PLAYER_TRIBUTE_DATA"] = 39] = "SEND_PLAYER_TRIBUTE_DATA";
})(TankTypes || (exports.TankTypes = TankTypes = {}));
var TileFlags;
(function (TileFlags) {
    TileFlags[TileFlags["FLAGS_TILEEXTRA"] = 1] = "FLAGS_TILEEXTRA";
    TileFlags[TileFlags["FLAGS_LOCKED"] = 2] = "FLAGS_LOCKED";
    TileFlags[TileFlags["FLAGS_SEED"] = 16] = "FLAGS_SEED";
    TileFlags[TileFlags["FLAGS_TREE"] = 25] = "FLAGS_TREE";
    TileFlags[TileFlags["FLAGS_FLIPPED"] = 32] = "FLAGS_FLIPPED";
    TileFlags[TileFlags["FLAGS_ROTATED_LEFT"] = 48] = "FLAGS_ROTATED_LEFT";
    TileFlags[TileFlags["FLAGS_OPEN"] = 64] = "FLAGS_OPEN";
    TileFlags[TileFlags["FLAGS_PUBLIC"] = 128] = "FLAGS_PUBLIC";
    TileFlags[TileFlags["FLAGS_SILENCED"] = 512] = "FLAGS_SILENCED";
    TileFlags[TileFlags["FLAGS_WATER"] = 1024] = "FLAGS_WATER";
    TileFlags[TileFlags["FLAGS_FIRE"] = 4096] = "FLAGS_FIRE";
    TileFlags[TileFlags["FLAGS_RED"] = 8192] = "FLAGS_RED";
    TileFlags[TileFlags["FLAGS_BLUE"] = 16384] = "FLAGS_BLUE";
    TileFlags[TileFlags["FLAGS_GREEN"] = 32768] = "FLAGS_GREEN";
    TileFlags[TileFlags["FLAGS_GLUE"] = 134217728] = "FLAGS_GLUE";
})(TileFlags || (exports.TileFlags = TileFlags = {}));
var TileOptionsFlags;
(function (TileOptionsFlags) {
    TileOptionsFlags[TileOptionsFlags["MUSIC_BLOCKS_DISABLED"] = 16] = "MUSIC_BLOCKS_DISABLED";
    TileOptionsFlags[TileOptionsFlags["MUSIC_BLOCKS_INVIS"] = 32] = "MUSIC_BLOCKS_INVIS";
})(TileOptionsFlags || (exports.TileOptionsFlags = TileOptionsFlags = {}));
var TilePropertiesFlags;
(function (TilePropertiesFlags) {
    TilePropertiesFlags[TilePropertiesFlags["MULTI_FACING"] = 1] = "MULTI_FACING";
    TilePropertiesFlags[TilePropertiesFlags["WRENCHABLE"] = 2] = "WRENCHABLE";
    TilePropertiesFlags[TilePropertiesFlags["PERMANENT"] = 4] = "PERMANENT";
    TilePropertiesFlags[TilePropertiesFlags["DROPLESS"] = 8] = "DROPLESS";
    TilePropertiesFlags[TilePropertiesFlags["NO_SEED"] = 16] = "NO_SEED";
    TilePropertiesFlags[TilePropertiesFlags["NO_SELF"] = 32] = "NO_SELF";
    TilePropertiesFlags[TilePropertiesFlags["NO_SHADOW"] = 64] = "NO_SHADOW";
    TilePropertiesFlags[TilePropertiesFlags["WORLD_LOCK"] = 128] = "WORLD_LOCK";
})(TilePropertiesFlags || (exports.TilePropertiesFlags = TilePropertiesFlags = {}));
var TileCategoryFlags;
(function (TileCategoryFlags) {
    TileCategoryFlags[TileCategoryFlags["BETA"] = 1] = "BETA";
    TileCategoryFlags[TileCategoryFlags["AUTO_PICKUP"] = 2] = "AUTO_PICKUP";
    TileCategoryFlags[TileCategoryFlags["MOD"] = 4] = "MOD";
    TileCategoryFlags[TileCategoryFlags["RANDOM_GROW"] = 8] = "RANDOM_GROW";
    TileCategoryFlags[TileCategoryFlags["PUBLIC"] = 16] = "PUBLIC";
    TileCategoryFlags[TileCategoryFlags["FOREGROUND"] = 32] = "FOREGROUND";
    TileCategoryFlags[TileCategoryFlags["HOLIDAY"] = 64] = "HOLIDAY";
    TileCategoryFlags[TileCategoryFlags["UNTRADEABLE"] = 128] = "UNTRADEABLE";
})(TileCategoryFlags || (exports.TileCategoryFlags = TileCategoryFlags = {}));
var TileActionTypes;
(function (TileActionTypes) {
    TileActionTypes[TileActionTypes["FIST"] = 0] = "FIST";
    TileActionTypes[TileActionTypes["WRENCH"] = 1] = "WRENCH";
    TileActionTypes[TileActionTypes["DOOR"] = 2] = "DOOR";
    TileActionTypes[TileActionTypes["LOCK"] = 3] = "LOCK";
    TileActionTypes[TileActionTypes["GEMS"] = 4] = "GEMS";
    TileActionTypes[TileActionTypes["TREASURE"] = 5] = "TREASURE";
    TileActionTypes[TileActionTypes["DEADLY_BLOCK"] = 6] = "DEADLY_BLOCK";
    TileActionTypes[TileActionTypes["TRAMPOLINE"] = 7] = "TRAMPOLINE";
    TileActionTypes[TileActionTypes["CONSUMABLE"] = 8] = "CONSUMABLE";
    TileActionTypes[TileActionTypes["ENTRANCE"] = 9] = "ENTRANCE";
    TileActionTypes[TileActionTypes["SIGN"] = 10] = "SIGN";
    TileActionTypes[TileActionTypes["SFX_WITH_EXTRA_FRAME"] = 11] = "SFX_WITH_EXTRA_FRAME";
    TileActionTypes[TileActionTypes["BOOMBOX"] = 12] = "BOOMBOX";
    TileActionTypes[TileActionTypes["MAIN_DOOR"] = 13] = "MAIN_DOOR";
    TileActionTypes[TileActionTypes["PLATFORM"] = 14] = "PLATFORM";
    TileActionTypes[TileActionTypes["BEDROCK"] = 15] = "BEDROCK";
    TileActionTypes[TileActionTypes["LAVA"] = 16] = "LAVA";
    TileActionTypes[TileActionTypes["FOREGROUND"] = 17] = "FOREGROUND";
    TileActionTypes[TileActionTypes["BACKGROUND"] = 18] = "BACKGROUND";
    TileActionTypes[TileActionTypes["SEED"] = 19] = "SEED";
    TileActionTypes[TileActionTypes["CLOTHES"] = 20] = "CLOTHES";
    TileActionTypes[TileActionTypes["FOREGROUND_WITH_EXTRA_FRAME"] = 21] = "FOREGROUND_WITH_EXTRA_FRAME";
    TileActionTypes[TileActionTypes["BACKGD_SFX_EXTRA_FRAME"] = 22] = "BACKGD_SFX_EXTRA_FRAME";
    TileActionTypes[TileActionTypes["BACK_BOOMBOX"] = 23] = "BACK_BOOMBOX";
    TileActionTypes[TileActionTypes["BOUNCY"] = 24] = "BOUNCY";
    TileActionTypes[TileActionTypes["POINTY"] = 25] = "POINTY";
    TileActionTypes[TileActionTypes["PORTAL"] = 26] = "PORTAL";
    TileActionTypes[TileActionTypes["CHECKPOINT"] = 27] = "CHECKPOINT";
    TileActionTypes[TileActionTypes["SHEET_MUSIC"] = 28] = "SHEET_MUSIC";
    TileActionTypes[TileActionTypes["ICE"] = 29] = "ICE";
    TileActionTypes[TileActionTypes["SWITCHEROO"] = 31] = "SWITCHEROO";
    TileActionTypes[TileActionTypes["CHEST"] = 32] = "CHEST";
    TileActionTypes[TileActionTypes["MAILBOX"] = 33] = "MAILBOX";
    TileActionTypes[TileActionTypes["BULLETIN"] = 34] = "BULLETIN";
    TileActionTypes[TileActionTypes["PINATA"] = 35] = "PINATA";
    TileActionTypes[TileActionTypes["DICE"] = 36] = "DICE";
    TileActionTypes[TileActionTypes["CHEMICAL"] = 37] = "CHEMICAL";
    TileActionTypes[TileActionTypes["PROVIDER"] = 38] = "PROVIDER";
    TileActionTypes[TileActionTypes["LAB"] = 39] = "LAB";
    TileActionTypes[TileActionTypes["ACHIEVEMENT"] = 40] = "ACHIEVEMENT";
    TileActionTypes[TileActionTypes["WEATHER_MACHINE"] = 41] = "WEATHER_MACHINE";
    TileActionTypes[TileActionTypes["SCORE_BOARD"] = 42] = "SCORE_BOARD";
    TileActionTypes[TileActionTypes["SUNGATE"] = 43] = "SUNGATE";
    TileActionTypes[TileActionTypes["PROFILE"] = 44] = "PROFILE";
    TileActionTypes[TileActionTypes["DEADLY_IF_ON"] = 45] = "DEADLY_IF_ON";
    TileActionTypes[TileActionTypes["HEART_MONITOR"] = 46] = "HEART_MONITOR";
    TileActionTypes[TileActionTypes["DONATION_BOX"] = 47] = "DONATION_BOX";
    TileActionTypes[TileActionTypes["TOYBOX"] = 48] = "TOYBOX";
    TileActionTypes[TileActionTypes["MANNEQUIN"] = 49] = "MANNEQUIN";
    TileActionTypes[TileActionTypes["SECURITY_CAMERA"] = 50] = "SECURITY_CAMERA";
    TileActionTypes[TileActionTypes["MAGIC_EGG"] = 51] = "MAGIC_EGG";
    TileActionTypes[TileActionTypes["GAME_RESOURCES"] = 52] = "GAME_RESOURCES";
    TileActionTypes[TileActionTypes["GAME_GENERATOR"] = 53] = "GAME_GENERATOR";
    TileActionTypes[TileActionTypes["XENONITE"] = 54] = "XENONITE";
    TileActionTypes[TileActionTypes["DRESSUP"] = 55] = "DRESSUP";
    TileActionTypes[TileActionTypes["CRYSTAL"] = 56] = "CRYSTAL";
    TileActionTypes[TileActionTypes["BURGLAR"] = 57] = "BURGLAR";
    TileActionTypes[TileActionTypes["COMPACTOR"] = 58] = "COMPACTOR";
    TileActionTypes[TileActionTypes["SPOTLIGHT"] = 59] = "SPOTLIGHT";
    TileActionTypes[TileActionTypes["WIND"] = 60] = "WIND";
    TileActionTypes[TileActionTypes["DISPLAY_BLOCK"] = 61] = "DISPLAY_BLOCK";
    TileActionTypes[TileActionTypes["VENDING_MACHINE"] = 62] = "VENDING_MACHINE";
    TileActionTypes[TileActionTypes["FISHTANK"] = 63] = "FISHTANK";
    TileActionTypes[TileActionTypes["PETFISH"] = 64] = "PETFISH";
    TileActionTypes[TileActionTypes["SOLAR"] = 65] = "SOLAR";
    TileActionTypes[TileActionTypes["FORGE"] = 66] = "FORGE";
    TileActionTypes[TileActionTypes["GIVING_TREE"] = 67] = "GIVING_TREE";
    TileActionTypes[TileActionTypes["GIVING_TREE_STUMP"] = 68] = "GIVING_TREE_STUMP";
    TileActionTypes[TileActionTypes["STEAMPUNK"] = 69] = "STEAMPUNK";
    TileActionTypes[TileActionTypes["STEAM_LAVA_IF_ON"] = 70] = "STEAM_LAVA_IF_ON";
    TileActionTypes[TileActionTypes["STEAM_ORGAN"] = 71] = "STEAM_ORGAN";
    TileActionTypes[TileActionTypes["TAMAGOTCHI"] = 72] = "TAMAGOTCHI";
    TileActionTypes[TileActionTypes["SWING"] = 73] = "SWING";
    TileActionTypes[TileActionTypes["FLAG"] = 74] = "FLAG";
    TileActionTypes[TileActionTypes["LOBSTER_TRAP"] = 75] = "LOBSTER_TRAP";
    TileActionTypes[TileActionTypes["ART_CANVAS"] = 76] = "ART_CANVAS";
    TileActionTypes[TileActionTypes["BATTLE_CAGE"] = 77] = "BATTLE_CAGE";
    TileActionTypes[TileActionTypes["PET_TRAINER"] = 78] = "PET_TRAINER";
    TileActionTypes[TileActionTypes["STEAM_ENGINE"] = 79] = "STEAM_ENGINE";
    TileActionTypes[TileActionTypes["LOCKBOT"] = 80] = "LOCKBOT";
    TileActionTypes[TileActionTypes["WEATHER_SPECIAL"] = 81] = "WEATHER_SPECIAL";
    TileActionTypes[TileActionTypes["SPIRIT_STORAGE"] = 82] = "SPIRIT_STORAGE";
    TileActionTypes[TileActionTypes["DISPLAY_SHELF"] = 83] = "DISPLAY_SHELF";
    TileActionTypes[TileActionTypes["VIP_ENTRANCE"] = 84] = "VIP_ENTRANCE";
    TileActionTypes[TileActionTypes["CHALLENGE_TIMER"] = 85] = "CHALLENGE_TIMER";
    TileActionTypes[TileActionTypes["CHALLENGE_FLAG"] = 86] = "CHALLENGE_FLAG";
    TileActionTypes[TileActionTypes["FISH_MOUNT"] = 87] = "FISH_MOUNT";
    TileActionTypes[TileActionTypes["PORTRAIT"] = 88] = "PORTRAIT";
    TileActionTypes[TileActionTypes["WEATHER_SPECIAL2"] = 89] = "WEATHER_SPECIAL2";
    TileActionTypes[TileActionTypes["FOSSIL"] = 90] = "FOSSIL";
    TileActionTypes[TileActionTypes["FOSSIL_PREP"] = 91] = "FOSSIL_PREP";
    TileActionTypes[TileActionTypes["DNA_MACHINE"] = 92] = "DNA_MACHINE";
    TileActionTypes[TileActionTypes["BLASTER"] = 93] = "BLASTER";
    TileActionTypes[TileActionTypes["VALHOWLA"] = 94] = "VALHOWLA";
    TileActionTypes[TileActionTypes["CHEMSYNTH"] = 95] = "CHEMSYNTH";
    TileActionTypes[TileActionTypes["CHEMTANK"] = 96] = "CHEMTANK";
    TileActionTypes[TileActionTypes["STORAGE"] = 97] = "STORAGE";
    TileActionTypes[TileActionTypes["OVEN"] = 98] = "OVEN";
    TileActionTypes[TileActionTypes["SUPER_MUSIC"] = 99] = "SUPER_MUSIC";
    TileActionTypes[TileActionTypes["GEIGER_CHARGER"] = 100] = "GEIGER_CHARGER";
    TileActionTypes[TileActionTypes["ADVENTURE_RESET"] = 101] = "ADVENTURE_RESET";
    TileActionTypes[TileActionTypes["TOMB_ROBBER"] = 102] = "TOMB_ROBBER";
    TileActionTypes[TileActionTypes["FACTION"] = 103] = "FACTION";
    TileActionTypes[TileActionTypes["RED_FACTION"] = 104] = "RED_FACTION";
    TileActionTypes[TileActionTypes["GREEN_FACTION"] = 105] = "GREEN_FACTION";
    TileActionTypes[TileActionTypes["BLUE_FACTION"] = 106] = "BLUE_FACTION";
    TileActionTypes[TileActionTypes["ANCES"] = 107] = "ANCES";
    TileActionTypes[TileActionTypes["FISHGOTCHI_TANK"] = 109] = "FISHGOTCHI_TANK";
    TileActionTypes[TileActionTypes["FISHING_BLOCK"] = 110] = "FISHING_BLOCK";
    TileActionTypes[TileActionTypes["ITEM_SUCKER"] = 111] = "ITEM_SUCKER";
    TileActionTypes[TileActionTypes["ITEM_PLANTER"] = 112] = "ITEM_PLANTER";
    TileActionTypes[TileActionTypes["ROBOT"] = 113] = "ROBOT";
    TileActionTypes[TileActionTypes["COMMAND"] = 114] = "COMMAND";
    TileActionTypes[TileActionTypes["TICKET"] = 115] = "TICKET";
    TileActionTypes[TileActionTypes["STATS_BLOCK"] = 116] = "STATS_BLOCK";
    TileActionTypes[TileActionTypes["FIELD_NODE"] = 117] = "FIELD_NODE";
    TileActionTypes[TileActionTypes["OUIJA_BOARD"] = 118] = "OUIJA_BOARD";
    TileActionTypes[TileActionTypes["ARCHITECT_MACHINE"] = 119] = "ARCHITECT_MACHINE";
    TileActionTypes[TileActionTypes["STARSHIP"] = 120] = "STARSHIP";
    TileActionTypes[TileActionTypes["AUTODELETE"] = 121] = "AUTODELETE";
    TileActionTypes[TileActionTypes["GREEN_FOUNTAIN"] = 122] = "GREEN_FOUNTAIN";
    TileActionTypes[TileActionTypes["AUTO_ACTION_BREAK"] = 123] = "AUTO_ACTION_BREAK";
    TileActionTypes[TileActionTypes["AUTO_ACTION_HARVEST"] = 124] = "AUTO_ACTION_HARVEST";
    TileActionTypes[TileActionTypes["AUTO_ACTION_HARVEST_SUCK"] = 125] = "AUTO_ACTION_HARVEST_SUCK";
    TileActionTypes[TileActionTypes["LIGHTNING_IF_ON"] = 126] = "LIGHTNING_IF_ON";
    TileActionTypes[TileActionTypes["PHASED_BLOCK"] = 127] = "PHASED_BLOCK";
    TileActionTypes[TileActionTypes["MUD"] = 128] = "MUD";
    TileActionTypes[TileActionTypes["ROOT_CUTTING"] = 129] = "ROOT_CUTTING";
    TileActionTypes[TileActionTypes["PASSWORD_STORAGE"] = 130] = "PASSWORD_STORAGE";
    TileActionTypes[TileActionTypes["PHASED_BLOCK_2"] = 131] = "PHASED_BLOCK_2";
    TileActionTypes[TileActionTypes["BOMB"] = 132] = "BOMB";
    TileActionTypes[TileActionTypes["WEATHER_INFINITY"] = 134] = "WEATHER_INFINITY";
    TileActionTypes[TileActionTypes["SLIME"] = 135] = "SLIME";
    TileActionTypes[TileActionTypes["UNK1"] = 136] = "UNK1";
    TileActionTypes[TileActionTypes["COMPLETIONIST"] = 137] = "COMPLETIONIST";
    TileActionTypes[TileActionTypes["UNK3"] = 138] = "UNK3";
    TileActionTypes[TileActionTypes["FEEDING_BLOCK"] = 140] = "FEEDING_BLOCK";
    TileActionTypes[TileActionTypes["KRANKENS_BLOCK"] = 141] = "KRANKENS_BLOCK";
    TileActionTypes[TileActionTypes["FRIENDS_ENTRANCE"] = 142] = "FRIENDS_ENTRANCE";
})(TileActionTypes || (exports.TileActionTypes = TileActionTypes = {}));
var TileExtraTypes;
(function (TileExtraTypes) {
    TileExtraTypes[TileExtraTypes["NONE"] = 0] = "NONE";
    TileExtraTypes[TileExtraTypes["DOOR"] = 1] = "DOOR";
    TileExtraTypes[TileExtraTypes["SIGN"] = 2] = "SIGN";
    TileExtraTypes[TileExtraTypes["LOCK"] = 3] = "LOCK";
    TileExtraTypes[TileExtraTypes["SEED"] = 4] = "SEED";
    TileExtraTypes[TileExtraTypes["MAILBOX"] = 6] = "MAILBOX";
    TileExtraTypes[TileExtraTypes["BULLETIN"] = 7] = "BULLETIN";
    TileExtraTypes[TileExtraTypes["DICE"] = 8] = "DICE";
    TileExtraTypes[TileExtraTypes["PROVIDER"] = 9] = "PROVIDER";
    TileExtraTypes[TileExtraTypes["ACHIEVEMENT"] = 10] = "ACHIEVEMENT";
    TileExtraTypes[TileExtraTypes["HEART_MONITOR"] = 11] = "HEART_MONITOR";
    TileExtraTypes[TileExtraTypes["DONATION_BLOCK"] = 12] = "DONATION_BLOCK";
    TileExtraTypes[TileExtraTypes["TOYBOX"] = 13] = "TOYBOX";
    TileExtraTypes[TileExtraTypes["MANNEQUIN"] = 14] = "MANNEQUIN";
    TileExtraTypes[TileExtraTypes["MAGIC_EGG"] = 15] = "MAGIC_EGG";
    TileExtraTypes[TileExtraTypes["GAME_RESOURCES"] = 16] = "GAME_RESOURCES";
    TileExtraTypes[TileExtraTypes["GAME_GENERATOR"] = 17] = "GAME_GENERATOR";
    TileExtraTypes[TileExtraTypes["XENONITE"] = 18] = "XENONITE";
    TileExtraTypes[TileExtraTypes["DESSUP"] = 19] = "DESSUP";
    TileExtraTypes[TileExtraTypes["CRYSTAL"] = 20] = "CRYSTAL";
    TileExtraTypes[TileExtraTypes["BURGLAR"] = 21] = "BURGLAR";
    TileExtraTypes[TileExtraTypes["SPOTLIGHT"] = 22] = "SPOTLIGHT";
    TileExtraTypes[TileExtraTypes["DISPLAY_BLOCK"] = 23] = "DISPLAY_BLOCK";
    TileExtraTypes[TileExtraTypes["VENDING_MACHINE"] = 24] = "VENDING_MACHINE";
    TileExtraTypes[TileExtraTypes["FISHTANK"] = 25] = "FISHTANK";
    TileExtraTypes[TileExtraTypes["SOLAR"] = 26] = "SOLAR";
    TileExtraTypes[TileExtraTypes["FORGE"] = 27] = "FORGE";
    TileExtraTypes[TileExtraTypes["GIVING_TREE"] = 28] = "GIVING_TREE";
    TileExtraTypes[TileExtraTypes["GIVING_TREE_STUMP"] = 29] = "GIVING_TREE_STUMP";
    TileExtraTypes[TileExtraTypes["STEAM_ORGAN"] = 30] = "STEAM_ORGAN";
    TileExtraTypes[TileExtraTypes["TAMAGOTCHI"] = 31] = "TAMAGOTCHI";
    TileExtraTypes[TileExtraTypes["SWING"] = 32] = "SWING";
    TileExtraTypes[TileExtraTypes["FLAG"] = 33] = "FLAG";
    TileExtraTypes[TileExtraTypes["LOBSTER_TRAP"] = 34] = "LOBSTER_TRAP";
    TileExtraTypes[TileExtraTypes["ART_CANVAS"] = 35] = "ART_CANVAS";
    TileExtraTypes[TileExtraTypes["BATTLE_CAGE"] = 36] = "BATTLE_CAGE";
    TileExtraTypes[TileExtraTypes["PET_TRAINER"] = 37] = "PET_TRAINER";
    TileExtraTypes[TileExtraTypes["STEAM_ENGINE"] = 38] = "STEAM_ENGINE";
    TileExtraTypes[TileExtraTypes["LOCKBOT"] = 39] = "LOCKBOT";
    TileExtraTypes[TileExtraTypes["WEATHER_SPECIAL"] = 40] = "WEATHER_SPECIAL";
    TileExtraTypes[TileExtraTypes["SPIRIT_STORAGE"] = 41] = "SPIRIT_STORAGE";
    TileExtraTypes[TileExtraTypes["UNKNOWN_1"] = 42] = "UNKNOWN_1";
    TileExtraTypes[TileExtraTypes["DISPLAY_SHELF"] = 43] = "DISPLAY_SHELF";
    TileExtraTypes[TileExtraTypes["VIP_ENTRANCE"] = 44] = "VIP_ENTRANCE";
    TileExtraTypes[TileExtraTypes["CHALLENGE_TIMER"] = 45] = "CHALLENGE_TIMER";
    TileExtraTypes[TileExtraTypes["CHALLENGE_FLAG"] = 46] = "CHALLENGE_FLAG";
    TileExtraTypes[TileExtraTypes["FISH_MOUNT"] = 47] = "FISH_MOUNT";
    TileExtraTypes[TileExtraTypes["PORTRAIT"] = 48] = "PORTRAIT";
    TileExtraTypes[TileExtraTypes["WEATHER_SPECIAL2"] = 49] = "WEATHER_SPECIAL2";
    TileExtraTypes[TileExtraTypes["FOSSIL_PREP"] = 50] = "FOSSIL_PREP";
    TileExtraTypes[TileExtraTypes["DNA_MACHINE"] = 51] = "DNA_MACHINE";
    TileExtraTypes[TileExtraTypes["MAGPLANT"] = 62] = "MAGPLANT";
    TileExtraTypes[TileExtraTypes["TESSERACT_MANIPULATOR"] = 69] = "TESSERACT_MANIPULATOR";
    TileExtraTypes[TileExtraTypes["GAIA_HEART"] = 70] = "GAIA_HEART";
    TileExtraTypes[TileExtraTypes["TECHNO_ORGANIC_ENGINE"] = 71] = "TECHNO_ORGANIC_ENGINE";
    TileExtraTypes[TileExtraTypes["KRANKEN_GALACTIC"] = 80] = "KRANKEN_GALACTIC";
    TileExtraTypes[TileExtraTypes["WEATHER_INFINITY"] = 77] = "WEATHER_INFINITY";
})(TileExtraTypes || (exports.TileExtraTypes = TileExtraTypes = {}));
